src/platform/nrfconnect/wifi/WiFiManager.cpp

```diff
@@ -208,9 +208,23 @@ CHIP_ERROR WiFiManager::Scan(const ByteSpan & ssid, ScanResultCallback resultCal
     mSsidFound          = false;
     mRecoveryArmed      = true;
 
-    if (net_mgmt(NET_REQUEST_WIFI_SCAN, iface, NULL, 0))
+    int ret = net_mgmt(NET_REQUEST_WIFI_SCAN, iface, NULL, 0);
+
+    if (ret)
     {
-        ChipLogError(DeviceLayer, "Scan request failed");
+        ChipLogError(DeviceLayer, "Scan request failed %d", ret);
+        if(mRecoveryActive){
+            /* Wi-Fi driver returned an error during recovery.
```

Comment:
I don't like this workaround for several reasons:
1. In theory the current state logic should be correct, so it would be better to investigate the issue in the supplicant code, instead of creating a workaround of solution that is already a workaround. Nevertheless I know we have to do something to fix this and we don't have much time, so let's say I'm ok with that.
2. We should at least try to analyze the returned error code to decide, whether we want to retry or not. Now we may fall into a loop of pointless retries, e.g. because the error code informs about something that will not be fixed just by waiting some time.
3. I don't think that the approach with incrementing the re-try time is a good idea for this particular case. Even if something is messed up with the supplicant state and we want to give it some more time to clean it, I think we should try to ask once again, but not more. If it wasn't fixed in a first few ms, I don't think it will be fixed ever. Maybe in case of getting error on a retry we should just crash and reboot the app to make sure we will not stuck in invalid networking state. Any thoughts @markaj-nordic?